shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/LedgerState.hs

```diff
@@ -1001,21 +1002,36 @@ updateNonMypopic nm rPot aps ss = nm
     aps' = Map.mapWithKey performance (aps `Map.union` performanceZero)
 
 
+-- | Compute the total stake
+totalStake
+  :: LedgerState crypto
+  -> Coin
+totalStake ls = totalUtxoStake + totalRewardStake
+  where
+    outputStake (TxOut (Addr _ (StakeRefBase _)) c) = c
+    outputStake (TxOut (Addr _ (StakeRefPtr _ )) c) = c
+    outputStake (TxOut (Addr _ (StakeRefNull  )) _) = 0
+    outputStake (TxOut (AddrBootstrap _)         _) = 0
+    (UTxO utxo) = _utxo . _utxoState $ ls
+    totalUtxoStake = sum $ Map.map outputStake utxo
```

Comment:
We're using a strict map here, so the sum will not fuse with the map (I'm pretty sure!). So it would be more efficient to use a fold here, given the UTxO is potentially quite large.

Reply:
good to know, thank for telling me! I'm going to update it right now